lhotse/dataset/vad.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+from math import isclose
+from typing import Dict, Optional
+
+import numpy as np
+import torch
+from torch.utils.data import Dataset
+
+from lhotse.cut import CutSet
+from lhotse.utils import Pathlike, Seconds
+
+EPS = 1e-8
+
+
+class VadDataset(Dataset):
+    """
+    The PyTorch Dataset for the voice activity detection task.
+    Returns a dict of:
+
+    .. code-block::
+
+        {
+            'features': (T x F) tensor
+            'is_voice': (T x 1) tensor
+        }
+
+    In the future, will be extended by graph supervisions.
+    """
+
+    def __init__(
+            self,
+            cuts: CutSet,
+            duration: Optional[Seconds] = 5.0,
+            root_dir: Optional[Pathlike] = None
+    ):
+        super().__init__()
+        self.cuts = cuts.cut_into_windows(duration, keep_excessive_supervisions=True).cuts
```

Comment:
```suggestion
        self.cuts = cuts.cut_into_windows(duration, keep_excessive_supervisions=True).filter(lambda cut: cut.duration == duration)
```

We need the filter step at the end to remove the last window for each recording which might be shorter (maybe `cut_into_windows` should do it automatically? I didn't want to impose my own preference on the users...)

Reply:
BTW another option is to call `.pad(duration)` instead to retain these windows.

Reply:
We can exclude them in `lhotse/cut.py`: https://github.com/pzelasko/lhotse/pull/64/commits/f5c166020d92ced2a0928fcd271067ddd73b4e57